src/frontend/src/expr/mod.rs

```diff
@@ -276,12 +276,21 @@ impl ExprImpl {
         Ok(backend_expr.eval_row(input).await?)
     }
 
-    /// Evaluate a constant expression.
-    pub fn eval_row_const(&self) -> RwResult<Datum> {
-        assert!(self.is_const());
-        self.eval_row(&OwnedRow::empty())
-            .now_or_never()
-            .expect("constant expression should not be async")
+    /// Try to evaluate an expression if it's a constant expression by `ExprImpl::is_const`.
+    ///
+    /// Returns...
+    /// - `None` if it's not a constant expression,
+    /// - `Some(Ok(_))` if constant evaluation succeeds,
+    /// - `Some(Err(_))` if there's an error while evaluating a constant expression.
+    pub fn eval_row_const(&self) -> Option<RwResult<Datum>> {
```

Comment:
To follow rust's api name convention, we should rename this to `try_eval_row_const`?

Reply:
how about `try_const_folding`?

Reply:
It is not very related to this PR. I will do it later.

Reply:
`try_fold_constant`

Reply:
Done